analysis-nicer-example.md

```diff
@@ -0,0 +1,314 @@
+
+<br>
+Heasoft higher than v6.31 is required in order to be able to run <code>nicerl3</code> tools.
+<br><br>
+
+<b>Running Outside Sciserver:</b><br>
+If running outside Sciserver, some changes will be needed, including:<br>
+&bull; Make sure heasoftpy and heasoft (higher than v6.31) are installed (<a href='https://heasarc.gsfc.nasa.gov/docs/software/lheasoft/'>Download and Install heasoft</a>).<br>
+&bull; Unlike on Sciserver, where the data is available locally, you will need to download the data to your machine.<br>
+</div>
+<!-- #endregion -->
+
+## 2. Module Imports
+We need the following python modules:
+
+<div style='color: #333; background: #ffffdf; padding:20px; border: 4px solid #fadbac'>
+Note that for heasoftpy < 1.4, <code>nicerl2</code> is accessed via <code>heasoftpy.nicerl2</code>. For heasoftpy >= 1.4, it is available under a separate module called <code>nicer</code> that needs to be imported explicitly.
+</div>
+
+```python
+## Import libraries that we will need.
+import os
+import sys
+from astropy.io import fits
+from astropy.table import Table
+import numpy as np
+import matplotlib.pylab as plt
+
+import heasoftpy as hsp
+import xspec as xs
+
+# import nicer tools from heasoftpy
+# for heasoftpy version >= 1.4, they are under heasoftpy.nicer.*
+# for heasoftpy version < 1.4, they are under heasoftpy.*
+try:
+    from heasoftpy.nicer import nicerl2, nicerl3_lc, nicerl3_spect
+except (ModuleNotFoundError, ImportError):
+    from heasoftpy import nicerl2, nicerl3_lc, nicerl3_spect
+```
+
+## 3. Set up the NICER obsid directory
+
+We are using OBSID `4142010107`. The data archive is mounted under `/FTP/..`. To find the exact location of the observation, we can use `pyvo` to query the archive using the VO services, or use Xamin, as illustrated in the [Getting Started](getting-started.md) and [Data Access](data-access.md) tutorials.
+
+Because `nicerl2` may modify of the observation directory, we copy it from the data location. If running the notebook outside Sciserver, the data will need to be downloaded from the archive.
+
+We will use the current directory as a working directory.
+
+```python
+nicerobsID = '4020180460'
+dataLocation = f'/FTP/nicer/data/obs/2021_12/{nicerobsID}'
+work_dir = os.getcwd()
+
+if not os.path.exists(nicerobsID):
+    os.system(f'cp -r {dataLocation} {work_dir}')
+```
+
+## 4. Process and Clean the Data.
+Next, we run the `nicerl2` pipeline to process and clean the data using `heasoftpy`
+
+There are different ways of calling a `heasoftpy` task. Here, we first create a dictionary that contains the input parameters for the `nicerl2` task, which is then passed to `nicer.nicerl2`
+
+```python
+# input
+inPars = {
+    'indir': nicerobsID,
+    'geomag_path': '/FTP/caldb/data/gen/pcf/geomag/',
+    'filtcolumns': 'NICERV4',
+    
+    'clobber': True, 
+    'noprompt': True,
+}
+
+# run the task
+out = nicerl2(inPars)
```

Comment:
This is giving me issues.  With the current test image, the heasoftpy version I get tells me that nicerl2 is a module and that for this to work, I'd have to call nicerl2.nicerl2.  So something's wrong in heasoftpy?

Reply:
The function itself is

`<function heasoftpy.nicer.nicerl2.nicerl2(args=None, **kwargs)>`

Reply:
This is a heasoftpy issue that I am aware of. It will be fixed, adn the image will be updated before release.

Reply:
Seems to be just nicer.  The other imports seem to work as expected, at least so far.